<a href="https://colab.research.google.com/github/mohandaz/HIA-302-Project/blob/main/HIA302_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

CLone Github Repository : https://github.com/mohandaz/HIA-302-Project

In [118]:
!git clone https://github.com/mohandaz/HIA-302-Project

fatal: destination path 'HIA-302-Project' already exists and is not an empty directory.


Libraries

In [119]:
import pandas as pd
import csv
import numpy as np

2.1 Demonstrate different ways to convert the horse-colic.txt file to horsecolic.csv.

In [ ]:
import pandas as pd
input_dir = '/content/HIA-302-Project/horse-colic.txt'
# Specify input and output file paths
output_dir = '/content/HIA-302-Project/horse-colic.csv'

# Read the text file using Pandas and write to CSV
horse_colic = pd.read_csv(input_dir)
horse_colic.to_csv(r"/content/HIA-302-Project/horse-colic.csv")

print(f"Conversion completed. CSV file saved at: {output_dir}")


Conversion completed. CSV file saved at: /content/HIA-302-Project/horse-colic.csv


Converting .txt to .csv - using Python built-in 'csv' module

In [ ]:
import csv

# Specify input and output file paths
input_dir = '/content/HIA-302-Project/horse-colic.txt'
output_dir = '/content/HIA-302-Project/horse-colic2.csv'

# Read the text file and write to CSV
with open(input_dir, 'r') as txt_file, open(output_dir, 'w', newline='') as csv_file:
    # Create CSV writer object
    csv_writer = csv.writer(csv_file)

    # Read each line from the text file and write to the CSV file
    for line in txt_file:
        # Split the line into a list of values using ',' as the delimiter
        values = line.strip().split(',')
        # Write the values to the CSV file
        csv_writer.writerow(values)

print(f"Conversion completed. CSV file saved at: {output_dir}")




Conversion completed. CSV file saved at: /content/HIA-302-Project/horse-colic2.csv


Checking data types

Question 2.2
 : Demonstrate that the csv file can read in Jupyter Notebook, and the dataset remains the same as the raw dataset.


In [ ]:
# Specify the path to the CSV file
csv_dir = '/content/HIA-302-Project/horse-colic.csv'

# Read the CSV file into a Pandas DataFrame
horse_colic = pd.read_csv(csv_dir)

# Display the first few rows of the CSV DataFrame

horse_colic.info()
horse_colic.head()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 298 entries, 0 to 297
Data columns (total 29 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       298 non-null    int64 
 1   1       298 non-null    object
 2   1.1     298 non-null    int64 
 3   534817  298 non-null    int64 
 4   39.2    298 non-null    object
 5   88      298 non-null    object
 6   20      298 non-null    object
 7   ?       298 non-null    object
 8   ?.1     298 non-null    object
 9   4       298 non-null    object
 10  1.2     298 non-null    object
 11  3       298 non-null    object
 12  4.1     298 non-null    object
 13  2       298 non-null    object
 14  ?.2     298 non-null    object
 15  ?.3     298 non-null    object
 16  ?.4     298 non-null    object
 17  4.2     298 non-null    object
 18  2.1     298 non-null    object
 19  50      298 non-null    object
 20  85      298 non-null    object
 21  2.2     298 non-null    object
 22  2.3     298 non-null    ob

0  1  1.1   534817   39.2   88  20  ? ?.1  4  ...     50    85 2.2   2.3  \
0  1  2    1   530334  38.30   40  24  1   1  3  ...  33.00  6.70   ?     ?   
1  2  1    9  5290409  39.10  164  84  4   1  6  ...  48.00  7.20   3  5.30   
2  3  2    1   530255  37.30  104  35  ?   ?  6  ...  74.00  7.40   ?     ?   
3  4  2    1   528355      ?    ?   ?  2   1  3  ...      ?     ?   ?     ?   
4  5  1    1   526802  37.90   48  16  1   1  1  ...  37.00  7.00   ?     ?   

  3.1 2.4  2208 0.1 0.2 2.5  
0   1   2     0   0   0   1  
1   2   1  2208   0   0   1  
2   2   2  4300   0   0   2  
3   1   2     0   0   0   2  
4   1   1  3124   0   0   2  

[5 rows x 29 columns]

2.3 Replase the missing values (i.e. with"?" with NaN (Not a Number) value in a loaded dataset using python

In [ ]:
horse_colic_with_NaN = horse_colic.replace("?","NaN")



In [ ]:
horse_colic_with_NaN.head()

0  1  1.1   534817   39.2   88   20    ?  ?.1  4  ...     50    85  2.2  \
0  1  2    1   530334  38.30   40   24    1    1  3  ...  33.00  6.70  NaN   
1  2  1    9  5290409  39.10  164   84    4    1  6  ...  48.00  7.20    3   
2  3  2    1   530255  37.30  104   35  NaN  NaN  6  ...  74.00  7.40  NaN   
3  4  2    1   528355    NaN  NaN  NaN    2    1  3  ...    NaN   NaN  NaN   
4  5  1    1   526802  37.90   48   16    1    1  1  ...  37.00  7.00  NaN   

    2.3 3.1 2.4  2208 0.1 0.2 2.5  
0   NaN   1   2     0   0   0   1  
1  5.30   2   1  2208   0   0   1  
2   NaN   2   2  4300   0   0   2  
3   NaN   1   2     0   0   0   2  
4   NaN   1   1  3124   0   0   2  

[5 rows x 29 columns]

2.4 Save the latest dataset with NaN as horse_colic_with_Nan.csv and ensure the data frame's row number is not saved into csv file

In [ ]:
output_dir = '/content/HIA-302-Project/horse_colic_with_NaN.csv'
# Save the DataFrame to a CSV file
horse_colic_with_NaN.to_csv(output_dir, index=False)

print(f"CSV file saved at: {output_dir}")


CSV file saved at: /content/HIA-302-Project/horse_colic_with_NaN.csv


In [ ]:
horse_colic_with_NaN.info()

2.5 Rename the header by following the description file (horse-colic.names) with not more than 10 characters i the header title

2.6 Save the latest dataset with header title as horse_colic_with_title.csv and ensure the data frame's row number is not saved into csv file.

2.7 Load the horse_colic_with_title.csv file and display list of total NaN at each column in the data set.

2.8 Visualize the totsl NaN at each column in the dataset with a bar chart and set colour to 'blue' using Metplotlib library

2.9 Retrive the datatypes for the dataset

2.10 In this dataset, will it be good idea to delete/drop tha NaN rows? Justify your answer

2.11 Read and evaluate the following codes in Figure 1. What method is the code using to reduce missing values from 1605 to 0? Justify your answer

2.12 Fill out the inline comments marked with hashtags # in lines 8,11,17,20,22. and 26. include the missing codes in line 5 and 6